core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -367,6 +385,13 @@ object SparkEnv extends Logging {
       metricsSystem,
       shuffleMemoryManager,
       conf)
+      
+    // Add a reference to tmp dir created by driver
+    if (isDriver) {
```

Comment:
Might call out why this is linked to the `isDriver` logic above for future readers. @JoshRosen thoughts now?